internal/providers/msentraid/msentraid_test.go

```diff
@@ -61,10 +61,16 @@ func TestVerifyUsername(t *testing.T) {
 
 		wantErr bool
 	}{
-		"Success when usernames are the same":   {requestedUsername: "foo@bar", authenticatedUser: "foo@bar"},
-		"Success when usernames differ in case": {requestedUsername: "foo@bar", authenticatedUser: "Foo@bar"},
-
-		"Error when usernames differ": {requestedUsername: "foo@bar", authenticatedUser: "bar@foo", wantErr: true},
+		"Success when usernames are the same":   {requestedUsername: "foo-bar@example", authenticatedUser: "foo-bar@example"},
```

Comment:
I'm a bit surprised that we didn't get the bug with `@` in the username before. Do you know what happened?

Reply:
The `.-_` in the regex was interpreted as "characters ranging from `.` to `_` which I assume must include `@` but not `-` (I didn't not check but that's the only explanation for the behavior we saw), so usernames which included a `@` were actually allowed but usernames which included a `-` were not.

Reply:
Oh, I see. So the issue was the `-` not the `@` as it was described in the revert PR?

Reply:
Well, both. The `@` was missing in the allowed characters. It was still allowed by accident because the `-` was not escaped so it was interpreted as a character range.